### PR TITLE
UCS/SYS/IB: Limit IB CQE size for specific ARM boards

### LINKED ISSUE #2025: data corruption on arm machine
caused by 7ced7a6
```
Note: Randomizing tests' orders with a seed of 5712 .
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from dcx/test_ucp_tag_xfer
[ RUN      ] dcx/test_ucp_tag_xfer.contig_exp/1
[     INFO ] 0 1581x10^1 500x10^2 [hpc-arm-hwi01:28522:0]   uct_iface.c:54   WARN: got active message id 0, but no handler installed
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2025

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6021/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls take a look

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -392,6 +392,22 @@ static ucs_status_t uct_ib_iface_init_lmc(uct_ib_iface_t *iface,
     return UCS_OK;
 }
 
+static int uct_ib_max_cqe_size()
+{
+    static int max_cqe_size = -1;
+
+    if (max_cqe_size == -1) {
+        max_cqe_size =
+#ifdef __aarch64__
```
Comment (shamisp) [inline src/uct/ib/base/ib_iface.c]: Is this really needed ? You detect the PCIe vendor and root complex. I'm not sure if this has to be arch specific

File: src/ucs/sys/sys.c [hunk 2]
```diff
@@ -807,6 +839,53 @@ double ucs_get_cpuinfo_clock_freq(const char *header, double scale)
...
+        goto out;
+    }
+
+    while (!found) {
+        entry = readdir(dir);
+        if (entry == NULL) {
+            goto out_closedir;
+        }
+
+        if (entry->d_name[0] == '.') {
```
Comment (shamisp) [inline src/ucs/sys/sys.c]: why do you need this check

Comment (yosefe) [reply src/ucs/sys/sys.c]: to avoid doing someting like "/../<name>"

File: src/uct/ib/base/ib_iface.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/base/ib_iface.c]: yes, i guess we can remove it

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6070/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6080/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6081/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6082/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6164/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6178/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6183/ for details (Mellanox internal link).

**Pull request merged**